src/sidebar/components/test/markdown-editor-test.js

```diff
@@ -0,0 +1,248 @@
+'use strict';
+
+const { createElement, render } = require('preact');
+const { act } = require('preact/test-utils');
+const { mount } = require('enzyme');
+
+const { LinkType } = require('../../markdown-commands');
+const MarkdownEditor = require('../markdown-editor');
+
+describe('MarkdownEditor', () => {
+  const formatResult = {
+    text: 'formatted text',
+    selectionStart: 0,
+    selectionEnd: 0,
+  };
+  const fakeMarkdownCommands = {
+    convertSelectionToLink: sinon.stub().returns(formatResult),
+    toggleBlockStyle: sinon.stub().returns(formatResult),
+    toggleSpanStyle: sinon.stub().returns(formatResult),
+    LinkType,
+  };
+
+  let MarkdownView;
+
+  beforeEach(() => {
+    fakeMarkdownCommands.convertSelectionToLink.resetHistory();
+    fakeMarkdownCommands.toggleBlockStyle.resetHistory();
+    fakeMarkdownCommands.toggleSpanStyle.resetHistory();
+
+    MarkdownView = function MarkdownView() {
+      return null;
+    };
+
+    MarkdownEditor.$imports.$mock({
+      '../markdown-commands': fakeMarkdownCommands,
+      './markdown-view': MarkdownView,
+    });
+  });
+
+  afterEach(() => {
+    MarkdownEditor.$imports.$restore();
+  });
+
+  const commands = [
+    {
+      command: 'Bold',
+      key: 'b',
+      effect: [fakeMarkdownCommands.toggleSpanStyle, '**', '**', 'Bold'],
+    },
+    {
+      command: 'Italic',
+      key: 'i',
+      effect: [fakeMarkdownCommands.toggleSpanStyle, '*', '*', 'Italic'],
+    },
+    {
+      command: 'Quote',
+      key: 'q',
+      effect: [fakeMarkdownCommands.toggleBlockStyle, '> '],
+    },
+    {
+      command: 'Insert link',
+      key: 'l',
+      effect: [fakeMarkdownCommands.convertSelectionToLink],
+    },
+    {
+      command: 'Insert image',
+      key: null,
+      effect: [
+        fakeMarkdownCommands.convertSelectionToLink,
+        fakeMarkdownCommands.LinkType.IMAGE_LINK,
+      ],
+    },
+    {
+      command: 'Insert math (LaTeX is supported)',
+      key: null,
+      effect: [
+        fakeMarkdownCommands.toggleSpanStyle,
+        '$$',
+        '$$',
+        'Insert LaTeX',
+      ],
+    },
+    {
+      command: 'Bulleted list',
+      key: 'u',
+      effect: [fakeMarkdownCommands.toggleBlockStyle, '* '],
+    },
+    {
+      command: 'Numbered list',
+      key: 'o',
+      effect: [fakeMarkdownCommands.toggleBlockStyle, '1. '],
+    },
+  ];
+  commands.forEach(({ command, key, effect }) => {
+    describe(`"${command}" toolbar command`, () => {
+      it('applies formatting when toolbar button is clicked', () => {
+        const onEditText = sinon.stub();
+        const wrapper = mount(
+          <MarkdownEditor
+            text="test"
+            readOnly={false}
```

Comment:
Erm...is this a valid prop for this component? I feel like I might have missed something!

Reply:
Oh, good catch. The `readOnly` prop has been removed.